src/map/lua/lua_baseentity.cpp

```diff
@@ -13189,11 +13189,15 @@ bool CLuaBaseEntity::addBardSong(CLuaBaseEntity* PEntity, uint16 effectID, uint1
  *  Function: charm()
  *  Purpose : Charms an entity target
  *  Example : mob:charm(target)
+ *  Notes: optional parameter specifying the duration in seconds
+ *         (default duration of 0 implies duration is controlled
+ *         by the status effect charm, which is the case for charmed players)
  ************************************************************************/
 
-void CLuaBaseEntity::charm(CLuaBaseEntity const* target)
+void CLuaBaseEntity::charm(CLuaBaseEntity const* target, sol::variadic_args va)
```

Comment:
If you are only going to have (always target, optional duration) you don't need to use variadic_args here, you can use a sol object I think? or a sol optional, or something like that. It makes the function signature say what the intended use is, rather than a catch-all

Reply:
Yeah, we use ` sol::object const& p0` for things that might be nil, but the function signature is mainly the same:
```
uint32 param0 = (p0 != sol::lua_nil) ? p0.as<uint32>() : 0;
```

Reply:
> Yeah, we use ` sol::object const& p0` for things that might be nil, but the function signature is mainly the same:
> 
> ```
> uint32 param0 = (p0 != sol::lua_nil) ? p0.as<uint32>() : 0;
> ```

Thanks for the feedback, I have fixed the PR.